en/advanced/out_of_tree_modules.md

```diff
@@ -1,80 +1,74 @@
-# Out-of-tree Modules
+# External Modules (Out-of-Tree)
 
-This describes the possibility to add an external module to the PX4 build.
+External modules provide a convenient mechanism for developers to manage/group proprietary modules that they want add to (or update in) PX4 firmware. External modules can use the same includes as internal modules and can interact with internal modules via uORB.
 
-External modules can use the same includes as internal modules and can interact
-with internal modules via uORB.
+This topic explains how to add an external ("out of tree") module to the PX4 build.
+
+> **Tip** We encourage you to contribute your changes into PX4, where possible!
 
 ## Usage
 
-- `EXTERNAL_MODULES_LOCATION` needs to point to a directory with the same
-  structure as Firmware (and thus contains a directory called `src`).
-- There are two options: copy an existing module (eg. examples/px4_simple_app)
-  to the external directory, or directly create a new module.
-- Rename the module (including `MODULE` in CMakeLists.txt) or remove it from the
-  existing Firmware cmake build config. This is to avoid conflicts with internal
-  modules.
-- Add a file `$EXTERNAL_MODULES_LOCATION/CMakeLists.txt` with content:
-
-```
-set(config_module_list_external
-    modules/<new_module>
-    PARENT_SCOPE
-    )
-```
-- add a line `EXTERNAL` to the `modules/<new_module>/CMakeLists.txt` within
-  `px4_add_module`, for example like this:
-
-```
-px4_add_module(
-	MODULE modules__test_app
-	MAIN test_app
-	STACK_MAIN 2000
-	SRCS
-		px4_simple_app.c
-	DEPENDS
-		platforms__common
-	EXTERNAL
-	)
-
-```
-
-
-# Out-of-tree uORB Message Definitions
-
-uORB messages can also be defined out-of-tree. For this, the
-`$EXTERNAL_MODULES_LOCATION/msg` needs to exist.
-
-- Place all new message definitions within the `$EXTERNAL_MODULES_LOCATION/msg`
-  directory. The format of these new out-of-tree message definitions are the same
-  as for any other [uORB message definition](../middleware/uorb.md#adding-a-new-topic).
+To create an external module:
+
+- Create a directory for external modules at the same level and with the same structure as Firmware (i.e. it must contain a directory called **src**). This is the *external directory* that we later refer to using `EXTERNAL_MODULES_LOCATION`.
```

Comment:
I don't quite understand what you mean with 'at the same level'

Reply:
I'm making an assumption here that the folder structure is:
```
Firmware
   /src
external_dir
   /src
```
It could of course be that external_dir can even be inside `Firmware/src`. Any rules for this?

Reply:
It can be anywhere, except inside Firmware, which does not make sense, since the reason to use external modules is to separate the trees (in terms of git and directory structure).

Reply:
Thanks. Fixed. I'll merge now :-)